specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-05-01-preview/machineLearningServices.json

```diff
@@ -0,0 +1,5247 @@
+        "computeType": {
+          "description": "The compute environment type for the service.",
+          "enum": [
+            "ACI",
+            "AKS"
+          ],
+          "type": "string",
+          "example": "AKS",
+          "x-ms-enum": {
+            "name": "ComputeEnvironmentType",
+            "modelAsString": true
+          }
+        },
+        "deploymentType": {
+          "description": "The deployment type for the service.",
+          "enum": [
+            "GRPCRealtimeEndpoint",
+            "HttpRealtimeEndpoint",
+            "Batch"
+          ],
+          "type": "string",
+          "example": "HttpRealtimeEndpoint",
+          "x-ms-enum": {
+            "name": "DeploymentType",
+            "modelAsString": false
+          }
+        },
+        "createdBy": {
+          "description": "The individual last responsible for creating or updating the service.",
+          "allOf": [
+            {
+              "$ref": "#/definitions/User"
+            }
+          ]
+        }
+      },
+      "discriminator": "computeType"
+    },
+    "PaginatedServiceList": {
+      "type": "object",
+      "description": "Paginated list of Machine Learning service objects wrapped in ARM resource envelope.",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "An array of Machine Learning compute objects wrapped in ARM resource envelope.",
+          "items": {
+            "$ref": "#/definitions/ServiceResponseBase"
+          }
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "A continuation link (absolute URI) to the next page of results in the list."
+        }
+      }
+    },
+    "User": {
+      "type": "object",
+      "properties": {
+        "userObjectId": {
+          "type": "string"
+        },
+        "userPuId": {
+          "type": "string"
+        },
+        "userIdp": {
+          "type": "string"
+        },
+        "userAltSecId": {
+          "type": "string"
+        },
+        "userIss": {
+          "type": "string"
+        },
+        "userTenantId": {
+          "type": "string"
+        },
+        "userName": {
+          "type": "string"
+        }
+      }
+    },
+    "ACIServiceResponse": {
+      "description": "The response for an ACI service.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ServiceResponseBase"
+        },
+        {
+          "type": "object",
+          "properties": {
+            "containerResourceRequirements": {
+              "description": "The container resource requirements.",
+              "allOf": [
+                {
+                  "$ref": "#/definitions/ContainerResourceRequirements"
+                }
+              ]
+            },
+            "scoringUri": {
+              "description": "The Uri for sending scoring requests.",
+              "type": "string"
+            },
+            "location": {
```

Comment:
Are services resources a tracked resource?  If not, what does "location" do here?  Can the location here be different than the parent workspace?

Reply:
Yes. When user submits a request to deploy his machine learning service  on Azure container instance(ACI), he also gets to  specify the azure region of the ACI, say "eastus2". Here, location refers to azure region.   

The location can be different than parent workspace and its a mandatory parameter.  These Azure container instances are unmanaged computes, and are provisioned on demand. Sometimes ACIs cant be provisioned in a region due to quota issues or for some other reasons,  and user can switch to  a different  region.